Makefile now attempts to pull repos.

insacuri opened the pull request:
Previously a repo would be skipped if had already been checked out.  Now it does a pull so that a 'make' will bring
all of the repos up to date.

I've tested it locally and it seems to work, but this should probably be tested on a few different machines to make sure it doesn't accidentally break anything

insacuri committed:
Makefile now attempts to pull repos.

Previously a repo would be skipped if had already been checked out.  Now it does a pull so that a 'make' will bring
all of the repos up to date

andrewmil reviewed (changes requested):
Need to update changelog.

Other than that it works on my system, definitely a useful change

insacuri committed:
Update changelog

warrenbailey reviewed (changes requested):
My concern with this change is you might have local changes which could conflict with a pull from remote and therefore break your environment.

I think this would be better as an optional command. I actually have two bash scripts which either update or reset all SDX repos. However they will still break if there's conflicts.

Maybe it would be better to add an optional target instead?

necrophonic commented on Makefile:
```diff
@@ -29,7 +29,8 @@ clone: check-env
 			git clone git@github.com:ONSdigital/${PREFIX}$${r}.git ${SDX_HOME}/${PREFIX}$${r}; \
 		else \
-			echo "  - already exists: skipping"; \
+			echo "Repo already cloned - pulling"; \
```
Could this actually be a `make update` rather than part of the `make` always? You may not want to pull on every build for reasons of speed or to avoid conflicts that you don't necessarily want to deal with yet.

insacuri committed:
Moved pulling functionality into it's own section

Having it git pull on every build can cause issues of speed and can cause conflicts.  Have a seperate
target that only updates all the repos is a fair compromise

insacuri committed:
Merge remote-tracking branch 'origin/master' into feature/pull-repos

JamesGardiner reviewed (changes requested):
One minor change I'd request - can we add --prune to the git pull command? This will stop dead branches collecting in my git repos :+1:

insacuri commented:
You couldn't just `git config --global remote.origin.prune true` yourself so that you won't ever again have to care about dead remote branches?

insacuri committed:
Merge remote-tracking branch 'origin/master' into feature/pull-repos
